studio-and-manager/docker-compose.yml

```diff
@@ -0,0 +1,249 @@
+      - KAFKA_PRODUCER_PASSWORD=${KAFKA_PRODUCER_PASSWORD}
+      - KAFKA_CONSUMER_USERNAME=${KAFKA_CONSUMER_USERNAME}
+      - KAFKA_CONSUMER_PASSWORD=${KAFKA_CONSUMER_PASSWORD}
+      - KAFKA_ENGINE_TOPIC=${KAFKA_ENGINE_TOPIC}
+      - AAD_APP_CLIENT_ID=${AAD_APP_CLIENT_ID}
+      - AAD_APP_TENANT_ID=${AAD_APP_TENANT_ID}
+      - ISSUER=${ISSUER}
+      - SERVER_HOST=${PWR_SERVER_HOST}
+      - OPENAI_API_KEY=${OPENAI_API_KEY}
+      - AZURE_OPENAI_API_KEY=${AZURE_OPENAI_API_KEY}
+      - AZURE_OPENAI_API_VERSION=${AZURE_OPENAI_API_VERSION}
+      - AZURE_OPENAI_ENDPOINT=${AZURE_OPENAI_ENDPOINT}
+      # Add all Azure OpenAI keys
+    depends_on:
+      - kafka
+      - postgres
+  studio:
+    build:
+      context: ../../PwR-Studio
+      dockerfile: ./studio/Dockerfile
+    restart: unless-stopped
+    environment:
+      - VITE_REACT_APP_AAD_APP_CLIENT_ID=${AAD_APP_CLIENT_ID}
+      - VITE_REACT_APP_AAD_APP_TENANT_ID=${AAD_APP_TENANT_ID}
+      - VITE_REACT_APP_AAD_APP_REDIRECT_URI=${AAD_APP_REDIRECT_URI}
+      - VITE_REACT_APP_AAD_APP_SCOPE_URI=${AAD_APP_SCOPE_URI}
+      - VITE_REACT_APP_SERVER_HOST=${STUDIO_SERVER_HOST}
+      - VITE_REACT_APP_INSIGHTS_KEY=${APPINSIGHTS_INSTRUMENTATIONKEY}
+      - VITE_REACT_EDITOR=code
+    ports:
+      - "4173:4173"
+    depends_on:
+      - server
+  engine:
+    image: jbengine:${JB_ENGINE_VERSION}
+    restart: unless-stopped
+    environment:
+      - KAFKA_BROKER=${KAFKA_BROKER}
+      - KAFKA_USE_SASL=${KAFKA_USE_SASL}
+      - KAFKA_PRODUCER_USERNAME=${KAFKA_PRODUCER_USERNAME}
+      - KAFKA_PRODUCER_PASSWORD=${KAFKA_PRODUCER_PASSWORD}
+      - KAFKA_CONSUMER_USERNAME=${KAFKA_CONSUMER_USERNAME}
+      - KAFKA_CONSUMER_PASSWORD=${KAFKA_CONSUMER_PASSWORD}
+      - KAFKA_ENGINE_TOPIC=${KAFKA_ENGINE_TOPIC}
+      - OPENAI_API_KEY=${OPENAI_API_KEY}
+      - AZURE_OPENAI_API_KEY=${AZURE_OPENAI_API_KEY}
+      - AZURE_OPENAI_API_VERSION=${AZURE_OPENAI_API_VERSION}
+      - AZURE_OPENAI_ENDPOINT=${AZURE_OPENAI_ENDPOINT}
+    depends_on:
+      - kafka
+
+  # Postgres and Kafka
+  postgres:
+    image: ankane/pgvector
+    ports:
+      - "5432:5432"
+    environment:
+      #   POSTGRES_USER: postgres
+      POSTGRES_PASSWORD: postgres
+    #   POSTGRES_DB: postgres
+    volumes:
+      - postgres_data:/var/lib/postgresql/data
+      - ./scripts/create-databases.sh:/docker-entrypoint-initdb.d/create-databases.sh
+      - ./jbmanager_dump.sql:/mnt/init/jbmanager_dump.sql
+      - ./jbstudio_dump.sql:/mnt/init/jbstudio_dump.sql
+
+  postgres-adminer:
+    image: adminer
+    restart: always
+    ports:
+      - 8080:8080
+  # metabase:
+  #   image: metabase/metabase:latest
+  #   ports:
+  #     - "9000:9000"
+  kafka:
+    image: docker.io/bitnami/kafka:3.6
+    ports:
+      - "9092:19092"
+    volumes:
+      - "kafka_data:/bitnami"
+      - ./scripts/create-all-topics.sh:/usr/bin/create-topics.sh
+    command: "sh -c './run.sh & /usr/bin/create-topics.sh && tail -f /dev/null'"
```

Comment:
In the latest version of JB Manager, create-topic reads topic names from env values